File: Cosmos-IOS/Cosmostation/Cosmostation/Dao/BnbTicker.swift
//
//  BnbTicker.swift
//  Cosmostation
//
//  Created by 정용주 on 2021/05/11.
//  Copyright © 2021 wannabit. All rights reserved.
//

import Foundation

public struct BnbTicker {
    var symbol: String?
    var baseAssetName: String?
    var quoteAssetName: String?
    var lastPrice: String?
    
    init(_ dictionary: NSDictionary?) {
        self.symbol = dictionary?["symbol"] as? String
        self.baseAssetName = dictionary?["baseAssetName"] as? String
        self.quoteAssetName = dictionary?["quoteAssetName"] as? String
        self.lastPrice = dictionary?["lastPrice"] as? String
    }
    
    func getLastPrice() -> NSDecimalNumber {
        return NSDecimalNumber.init(string: lastPrice)
    }
}

File: Cosmos-Android/app/src/main/java/wannabit/io/cosmostaion/dao/BnbTicker.java
package wannabit.io.cosmostaion.dao;

import com.google.gson.annotations.SerializedName;

public class BnbTicker {
    @SerializedName("symbol")
    public String symbol;

    @SerializedName("baseAssetName")
    public String baseAssetName;

    @SerializedName("quoteAssetName")
    public String quoteAssetName;

    @SerializedName("lastPrice")
    public String lastPrice;
}
